lib/plugins/aws/deploy/lib/createStack.js

```diff
@@ -40,7 +41,17 @@ module.exports = {
       'CloudFormation',
       'createStack',
       params
-    ).then((cfData) => this.monitorStack('create', cfData));
+    ).then((cfData) => this.monitorStack('create', cfData)).catch((e) => {
+      if (this.serverless.service.deployment && this.serverless.service.deployment.deploymentId) {
+        const deploymentData = this.serverless.service.deployment;
+        deploymentData.status = 'Failed';
+        return platform.createDeployment(deploymentData)
```

Comment:
Shouldn't this be `updateDeployment` and the `createDeployment` is called before the CloudFormation template is even handed over to AWS?